app/src/main/java/ai/elimu/launcher/ui/HomeScreensActivity.java

```diff
@@ -152,6 +158,30 @@ public void onPageScrollStateChanged(int state) {
         Timber.i("applications.size(): " + applications.size());
     }
 
+    private void checkIfAppstoreIsInstalled() {
+        try {
+            PackageInfo packageInfoAppstore = getPackageManager().getPackageInfo(BuildConfig.APPSTORE_APPLICATION_ID, 0);
+            Timber.i("packageInfoAppstore.versionCode: " + packageInfoAppstore.versionCode);
+        } catch (PackageManager.NameNotFoundException e) {
+            Timber.w(null, e);
+            new MaterialDialog.Builder(this)
+                    .content(getResources().getString(R.string.appstore_needed) + BuildConfig.APPSTORE_APPLICATION_ID)
+                    .positiveText(getResources().getString(R.string.download))
+                    .onPositive(new MaterialDialog.SingleButtonCallback() {
+                        @Override
+                        public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
+                            goToAppstoreDownload();
+                        }
+                    })
+                    .show();
+        }
+    }
+
+    private void goToAppstoreDownload() {
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(APPSTORE_DOWNLOAD_URL));
```

Comment:
Maybe move this code into `onClick` (line 173), to reduce the number of methods?